src/lib/outgoing-channel.js

```diff
@@ -112,43 +104,46 @@ module.exports = class OutgoingChannel extends EventEmitter2 {
     // this will complain if the amount exceeds the maximum
     yield this._balance.add(transfer.amount)
     const claim = yield this._balance.get()
-
-    if (this._balance.getMax().mul(this._fundPercent).lt(claim)) {
-      console.log('the new outgoing balance is greater than the threshold')
-      yield this._fundChannel()
+    const threshold = this._balance.getMax().mul(this._fundPercent)
+
+    if (threshold.lt(claim)) {
+      debug('balance of', claim.toString(), 'exceeds threshold',
+        threshold.toString(), '. triggering fund tx.')
+      yield co.wrap(this._fundChannel).call(this).catch((e) => {
+        console.error(e)
```

Comment:
:+1: I'm going to do a lot of cleaning up when I refactor this code to use payment channel framework